pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/TransformOperand.java

```diff
@@ -18,26 +18,123 @@
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) == 0;
+            } else {
+              return ((Comparable) _lhs.apply(row)).compareTo(_rhs.apply(row)) == 0;
+            }
+          }
+        };
+      case "notEquals":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
+            if (_requireCasting) {
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) != 0;
+            } else {
+              return ((Comparable) _lhs.apply(row)).compareTo(_rhs.apply(row)) != 0;
+            }
+          }
+        };
+      case "greaterThan":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
+            if (_requireCasting) {
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) > 0;
+            } else {
+              return ((Comparable) _lhs.apply(row)).compareTo(_rhs.apply(row)) > 0;
+            }
+          }
+        };
+      case "greaterThanOrEqual":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
+            if (_requireCasting) {
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) >= 0;
+            } else {
+              return ((Comparable) _lhs.apply(row)).compareTo(_rhs.apply(row)) >= 0;
+            }
+          }
+        };
+      case "lessThan":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
+            if (_requireCasting) {
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) < 0;
+            } else {
+              return ((Comparable) _lhs.apply(row)).compareTo(_rhs.apply(row)) < 0;
+            }
+          }
+        };
+      case "lessThanOrEqual":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
+            if (_requireCasting) {
+              return ((Comparable) FunctionInvokeUtils.convert(_lhs.apply(row), _commonCastType)).compareTo(
+                  FunctionInvokeUtils.convert(_rhs.apply(row), _commonCastType)) <= 0;
```

Comment:
(nit) may want to consider making this a little less short-hand for the code to be more readable.